File: lib/models/order/OrderPlacementQueryDto.ts
import OrderItemQueryDto from "./OrderItemQueryDto";
import OrderQueryDto from "./OrderQueryDto";

interface IOrderPlacementQueryDto {
    order: OrderQueryDto;
    orderItems: OrderItemQueryDto[];
}

class OrderPlacementQueryDto implements IOrderPlacementQueryDto {
    order: OrderQueryDto;
    orderItems: OrderItemQueryDto[];

    constructor(order: OrderQueryDto, orderItems: OrderItemQueryDto[]) {
        this.order = order;
        this.orderItems = orderItems;
    }
}

export default OrderPlacementQueryDto;

File: Application/Dtos/OrderHandling/OrderPlacementQueryDto.cs
using Application.Dtos.Order;
using Application.Dtos.OrderItem;

namespace Application.Dtos.OrderHandling
{
    public class OrderPlacementQueryDto
    {
        public OrderQueryDto           order      { get; set; }
        public List<OrderItemQueryDto> orderItems { get; set; }
    }
}